tests/scripts/uniswap.v2.test.js

```diff
@@ -104,29 +104,27 @@ async function ethTests() {
   console.log(allowances);
 
   for (let token of tokens) {
-    if (parseFloat(allowances[token]) < 1000.0) {
-      // call /approve on each token
-      console.log(`Approving 5000 ${token}...`);
-      let approve1 = await request('post', '/eth/approve', {
-        token: token,
+    // call /approve on each token
+    console.log(`Resetting allowance for ${token} to 5000...`);
+    let approve1 = await request('post', '/eth/approve', {
+      token: token,
+      connector: 'uniswap',
+      amount: '5000',
+    });
+    console.log(approve1);
+    while (allowances[token] !== approve1.amount) {
```

Comment:
Here's what I get when running this below.

From Etherscan, it looks like the 2nd approval transaction isn't actually called. Since we plan to improve handling of batch transactions in V2, let's not test for this now. 

Instead, please change this test so that it just checks whether the `/approve` route works and just check 1 of the 2 tokens.

```
Starting ETH tests
***************************************************
Token symbols used in tests:  [ 'WETH', 'DAI' ]
Checking status of gateway server...
ok
Checking balances...
{
  ETH: '2.154760387468044644',
  DAI: '88.093140718715770165',
  WETH: '0.344455115432178346'
}
calling balances with invalid token symbols ABC and XYZ...
{ ETH: '2.154760387468044644' }
checking initial allowances...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Resetting allowance for WETH to 5000...
{
  network: 'kovan',
  timestamp: 1629322258711,
  latency: 2.245,
  tokenAddress: '0xd0A1E359811322d97991E03f863a0C30C2cF029C',
  spender: '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
  amount: '5000.000000000000000000',
  approval: {
    type: 2,
    chainId: 42,
    nonce: 93,
    maxPriorityFeePerGas: { type: 'BigNumber', hex: '0x153005ce00' },
    maxFeePerGas: { type: 'BigNumber', hex: '0x153005ce00' },
    gasPrice: null,
    gasLimit: { type: 'BigNumber', hex: '0x0186a0' },
    to: '0xd0A1E359811322d97991E03f863a0C30C2cF029C',
    value: { type: 'BigNumber', hex: '0x00' },
    data: '0x095ea7b30000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d00000000000000000000000000000000000000000000010f0cf064dd59200000',
    accessList: [],
    hash: '0xa85df99ca8ee3bdbeb67814997ceb178911b60f7b52020b7397e4af0e16de4b0',
    v: 1,
    r: '0xcc12a0de43fa062283fc67f4800bc81c850d30a06870d8babec425fc94824981',
    s: '0x0496a158683fe38c59a2548e0c2a047b54510c3fac7c9fc9778b8aad39db6940',
    from: '0x9FA3156B802eA7ECFe55173Eafc296f509a28777'
  }
}
Resetting allowance for DAI to 5000...
{
  network: 'kovan',
  timestamp: 1629322260962,
  latency: 2.789,
  tokenAddress: '0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa',
  spender: '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
  amount: '5000.000000000000000000',
  approval: {
    type: 2,
    chainId: 42,
    nonce: 93,
    maxPriorityFeePerGas: { type: 'BigNumber', hex: '0x153005ce00' },
    maxFeePerGas: { type: 'BigNumber', hex: '0x153005ce00' },
    gasPrice: null,
    gasLimit: { type: 'BigNumber', hex: '0x0186a0' },
    to: '0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa',
    value: { type: 'BigNumber', hex: '0x00' },
    data: '0x095ea7b30000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d00000000000000000000000000000000000000000000010f0cf064dd59200000',
    accessList: [],
    hash: '0x4f63c8c2f5236a0f410a826988ad8478f11a28d52978ed514d1dde96d2b193d6',
    v: 1,
    r: '0x199c5cea7b2fe43b50fa00fb9c33194def628f7c079f064e24061bc428fded8b',
    s: '0x6e206d326f5eadfa9efc81b4eb22da48d02c1f9f248e98d45e4ba0672195a701',
    from: '0x9FA3156B802eA7ECFe55173Eafc296f509a28777'
  }
}
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ DAI: '1000.000000000000000000', WETH: '5000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
Rechecking allowances to confirm approval...
{ WETH: '5000.000000000000000000', DAI: '1000.000000000000000000' }
Waiting for atleast 1 block time (i.e 13 secs) to give time for approval to be mined.
```

Reply:
The reason the second approval fails is because it has the same nonce as the first approval. So it was dropped because the first tx got mined.